### PR TITLE
Add support for the NestedForm form widget

### PR DESCRIPTION
**Pull request by LukeTowers**
Fixes https://github.com/octobercms/october/issues/5111. WIP, not finished yet.

### CODE REVIEW

File: formwidgets/MLNestedForm.php [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        $this->initLocale();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function render()
+    {
+        $this->isAvailable = Locale::isAvailable();
+
```
Comment (mjauvin) [inline formwidgets/MLNestedForm.php:40]:
```suggestion
```
This is done in initLocale() called in init(). Shouldn't be required here.

**Comment by iboved** [comment]
@LukeTowers what is not finished yet? Can you answer please, I will try to finish this, I need this feature. Thanks!

**Comment by daftspunk** [comment]
I don't think you'll be getting an answer from Luke, @iboved. Have you tested this PR in its current state?

**Comment by iboved** [comment]
@daftspunk some methods are missing in the MLNestedForm or NestedForm classes - [processSaveValue](https://github.com/rainlab/translate-plugin/pull/584/files#diff-c2f342249b41720fcef2b17f86801e6006272c04ee121174aa91e542a7f885d5R159) and [processItems](https://github.com/rainlab/translate-plugin/pull/584/files#diff-c2f342249b41720fcef2b17f86801e6006272c04ee121174aa91e542a7f885d5R148).
Luke seems to have copied MLRepeater which explain these methods being used.

**Comment by hwinne** [comment]
@daftspunk Hey Sam, is there a possibility this PR will be finished and added in the near future?

**Comment by daftspunk** [comment]
> Luke seems to have copied MLRepeater which explain these methods being used.

Confirmed this is just a carbon copy of the MLRepeater with some fields renamed. Closing to replace with something that works.

**Comment by daftspunk** [comment]
This original implementation was almost there. Complete support has been added in v1.10.0

@hwinne @iboved Please update to v1.10.0 to test this. For support: https://octobercms.com/contact

Thanks

**Comment by hwinne** [comment]
@daftspunk It's not working for me. 

I tried adding `translatable: true` to a nestedform in a Settings model and a nestedform in a Syntax field (where a repeater field translatable is working fine).

## Versions
October CMS: 2.2.1
RainLab Translate: 1.10.1

**Comment by daftspunk** [comment]
Hey @hwinne,

Could you try making this change to see if it helps? https://github.com/rainlab/pages-plugin/commit/40f7d02b29dfb2369c66ecd55f77088b7f905027

**Comment by hwinne** [comment]
Hi @daftspunk,

Thank you for your response.

I added that, but then I get an error, which I can not get away.
image

I removed all the contents from my layout, but i still get the error.
Any idea where it could come from?

**Comment by hwinne** [comment]
Yes! That did it for me.

It works fine when creating a new page, but:
I had a problem with opening existing pages though, I do not know if thats on me or not.

My pages contain content blocks with nested forms.

This is the error I got:
image

**Comment by daftspunk** [comment]
Ok let me know what field configuration you are using and I'll take a look.

**Comment by daftspunk** [comment]
Thanks

- Added a test here: https://github.com/daftspunk/test-theme/commit/dc2e046e3a35a8d89a762a2cf8b5419b6658992c
- Moving this to a new issue

**Comment by daftspunk** [comment]
Moved to #665

**Pull request closed without merging**